src/gluonts/model/gp_forecaster/_network.py

```diff
@@ -208,12 +219,20 @@ def hybrid_forward(
             sigma=sigma,
             kernel=kernel,
             context_length=self.context_length,
+            prediction_length=self.prediction_length,
             ctx=self.ctx,
             float_type=self.float_type,
             max_iter_jitter=self.max_iter_jitter,
             jitter_method=self.jitter_method,
         )
-        return gp.log_prob(past_time_feat, past_target)
+        log_prob_data = gp.log_prob(past_time_feat, past_target)
+        log_prob_forecast = gp.log_prob_predictive(
+            past_time_feat, past_target, future_time_feat, future_target
+        )
+        if self.train_forecast:
```

Comment:
I will you can make this part a bit more succinct with
`return log_prob_data + self.train_forecast * log_prob_forecast`, since `self.train_forecast` is a boolean here this should work.

Reply:
Actually, we don't need to compute `log_prob_predictive` if `self.train_forecast = True`.  so we can have:
` if self.train_forecast: return 
`